contribute/e2e/CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+- Raise/fix issues on the e2e, CI documentation
+
+## GENERAL CONTRIBUTION GUIDELINES
+
+Here are some general contribution guidelines. 
+
+### Adding new tests/test-categories
+
+To add new tests, please raise an issue describing the objective, pre-requisites, test steps, expected results & cleanup routines.
+In case of proposals to add new test sections/categories, also add a generic description explaining the purpose and value addition
+
+Once accepted, the issue can be assigned to interested contributors who can make relevant changes with a pull request. 
+
+### Create new application YAMLs
+
+To add new applications to e2e, please consider the points made here, as applicable : 
+
+- https://kubernetes.io/docs/concepts/configuration/overview/
+- https://www.mirantis.com/blog/introduction-to-yaml-creating-a-kubernetes-deployment/
+- https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/
+
+These can be placed in here : https://github.com/openebs/openebs/tree/master/k8s/demo
+
+### Create ansible playbooks to execute test workflows
+
+The test playbooks can be placed in e2e/ansible/playbooks/<test-category>. Here are some considerations 
+
+- Each test playbook is recommended to be constructed with the following template.
+
+  The main test playbook, which statically "includes" following files :
+
+  - Test variables YAML
+  - Pre-requisites tasks YAML
+  - Cleanup tasks YAML
+  
+  Auxiliary scripts that are invoked in the test playbook can be placed in the same directory. 
+
+  A sample test playbook can be seen [here](https://github.com/openebs/openebs/tree/master/e2e/ansible/playbooks/hyperconverged/test-k8s-percona-mysql-pod)
+
+- Test playbooks that deploy applications are recommended to include steps to generate application-specfic workloads that
+  will exercise the OpenEBS storage volume. 
+  
+- Here are some ansible best practices while constructing playbooks : 
+ 
+  - https://www.jeffgeerling.com/blog/yaml-best-practices-ansible-playbooks-tasks 
+  
+  The OpenEBS ansible environment setup is detailed [here](https://github.com/openebs/openebs/blob/master/e2e/ansible/openebs-on-premise-deployment-guide.md)
+  
+### Submit changes to fix bugs/improve e2e tests 
+
+Please explain the fix made in detail and add the verbose output of the test playbook run with the fixes (Run the playbook with a -v option for this). Also, do consider the general ansible playbook best practices referenced in the previous section.
+
+### Raise/fix issues on the CI framework 
+
+As with other issues, please mention the bug/enhancements in detail with logs/references if possible, respectively. 
+
+The jenkins configuration used to setup the OpenEBS e2e can be found [here](https://github.com/openebs/openebs/blob/master/e2e/jenkins/README.md)
+
+### Raise/fix issues on the e2e, CI documentation
+
+Getting Documentation Right is Hard! Please raise a PR with you proposed changes to the READMEs, tutorials, deployment guides etc.,

```

Comment:
etc. = and so on.

Reply:
restricted to these 3 - readme, tutorial and deployment guides.